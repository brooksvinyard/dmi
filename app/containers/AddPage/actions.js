/*
 *
 * AddPage actions
 *
 */

import {
  CHANGE_STRING,
  POST_STRINGS,
  POST_STRINGS_SUCCESS,
  POST_STRINGS_ERROR,
} from './constants';

export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

export function postString() {
  return {
    type: POST_STRINGS,
  };
}

export function postStringSuccess() {
  return {
    type: POST_STRINGS_SUCCESS,
  };
}

export function postStringError(error) {
  return {
    type: POST_STRINGS_ERROR,
    error,
  };
}
