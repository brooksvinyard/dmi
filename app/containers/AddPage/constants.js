/*
 *
 * AddPage constants
 *
 */

export const CHANGE_STRING = 'dmi/AddPage/CHANGE_STRING';
export const POST_STRINGS = 'dmi/AddPage/POST_STRINGS';
export const POST_STRINGS_SUCCESS = 'dmi/AddPage/POST_STRINGS_SUCCESS';
export const POST_STRINGS_ERROR = 'dmi/AddPage/POST_STRINGS_ERROR';
