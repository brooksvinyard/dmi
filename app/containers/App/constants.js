/*
 * AppConstants
 */

export const LOAD_STRINGS = 'dmi/App/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'dmi/App/LOAD_STRINGS_SUCCESS';
export const LOAD_STRINGS_ERROR = 'dmi/App/LOAD_STRINGS_ERROR';
