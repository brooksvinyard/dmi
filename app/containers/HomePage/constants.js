/*
 * HomePageConstants
 */

export const GET_STRINGS = 'dmi/HomePage/GET_STRINGS';
