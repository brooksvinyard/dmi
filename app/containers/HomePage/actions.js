/*
 * Home Actions
 *
 */

import { GET_STRINGS } from './constants';

/**
 * Gets the strings from the database
 *
 * @return {object} An action object with a type of GET_STRINGS
 */
export function getStrings() {
  return {
    type: GET_STRINGS,
  };
}
