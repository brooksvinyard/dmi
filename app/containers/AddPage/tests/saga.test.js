/**
 * Tests for AddPage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { POST_STRINGS } from '../constants';
import { postStringSuccess, postStringError } from '../actions';

import addPageSaga, { postString } from '../saga';

const string = 'Hello World';

/* eslint-disable redux-saga/yield-effects */
describe('postString Saga', () => {
  let postStringGenerator;

  beforeEach(() => {
    postStringGenerator = postString();

    const selectDescriptor = postStringGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = postStringGenerator.next(string).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the postStringSuccess action if it requests the data successfully', () => {
    const response = [
      {
        string: 'Hello World',
      },
    ];
    const putDescriptor = postStringGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(postStringSuccess(response, string)));
  });

  it('should call the postStringError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = postStringGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(postStringError(response)));
  });
});

describe('githubDataSaga Saga', () => {
  const addPageSagaSaga = addPageSaga();

  it('should start task to watch for LOAD_STRINGS action', () => {
    const takeLatestDescriptor = addPageSagaSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(POST_STRINGS, postString));
  });
});
