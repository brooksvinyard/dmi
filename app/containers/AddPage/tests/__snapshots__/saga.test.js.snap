// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postString Saga should call the postStringError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`postString Saga should call the postStringError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api/v1/strings/",
      Object {
        "body": "{\\"string\\":\\"Hello World\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`postString Saga should dispatch the postStringSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`postString Saga should dispatch the postStringSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api/v1/strings/",
      Object {
        "body": "{\\"string\\":\\"Hello World\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
